### DISCUSSION TITLE
pytest integration

### DISCUSSION THREAD

**Discussion by kynan** [prysk/prysk#128] [Q&A] [answered]
I'm currently using `cram` in [nbstripout](https://github.com/kynan/nbstripout) via [pytest](https://docs.pytest.org), thanks to [pytest-cram](https://github.com/tbekolay/pytest-cram).

I was considering changing `nbstripout` to use `prysk` but I'd like to keep using pytest. I might fork pytest-cram and adapt it to use prysk but wanted to check if you had plans for pytest integration already?

**Comment by Nicoretti** [comment]
Hi @kynan,

happy to hear you have some interest in using **_prysk_**.
Generally I think a **_pytest_** plugin for **_prysk_** would be nice.

To be transparent about the current state and _"roadmap"_ of the project I must say that
in the recent months I had some stuff coming up which prevented me from
investing lots of time into **_prysk_** (I hope this will change soon).

Still my personal next _"go to topic"_ is a refactoring of the test(s) execution, so
multiple test files can be executed in parallel using the `multiprocessing` module.

That said, I would be open to make a **_pytest_** plugin part of the **_prysk_** project
itself, e.g. via feature flag or just as a additional package within the project.
(If that makes sense)

Regarding it's implementation though I don't see me implementing it any time soon,
but I would be more than willing/happy to enable you or anyone else working on it.
(by adding project collaborators to **GH** and **PyPi** etc.)

best
Nico

**Reply by kynan** [reply]
Thanks! It may make sense to keep it as a separate project in the spirit of not shipping functionality not everyone will use.

I'll let you know if I end up finding time to work on this - also not sure I'll get round to it anytime soon :)

**Reply by Nicoretti** [reply]
Hi @kynan,

just wanted to let you know that I finally ended up working on this (see [PR-161](#161)) after consolidating the project (tickets, PRs etc). I think executing prysk tests in parallel using [gnu-parallel](https://www.gnu.org/software/parallel/) or [pytest](https://docs.pytest.org/en/7.1.x/contents.html) together with [pytest-xdist](https://github.com/pytest-dev/pytest-xdist) once the `pytest-prysk` plugin is available, is good enough or maybe even the better solution.

**ℹ️ Status**
So far I was mainly tinkering to understand the pytest plugin mechansim better. I still need to do some cleanup and add tests & documentation. Anyhow I am confident I will release it the next couple of weeks (1-2 🤞).

If this feature is still relevant to you feel free to subscribe to the [PR-161](#161).

best
Nico

**Comment by Nicoretti** [comment, marked as answer]
A pytest plugin have been added in version [0.13.0](https://pypi.org/project/prysk/0.13.0/). For further details checkout the [pytest-plugin section](https://www.prysk.net/0.13.0/pytest-plugin.html) of the documentation.
